tests/Unit/Traits/HasJsonModelAttributesTest.php

```diff
@@ -424,4 +425,23 @@ public function testAttributeCanBeTestedWithIsset()
         self::assertFalse($model->jsonModel->a ?? false);
         self::assertFalse($model->jsonModel->other ?? false);
     }
+
+    public function testAttributeIsResetByRefresh()
+    {
+        $model = mock(EventedModelWithJsonAttributes::class)->makePartial()->shouldAllowMockingProtectedMethods();
+        // We're mocking the guts of Model::refresh to not actually require a database
+        $model->shouldReceive('setKeysForSelectQuery')->once()->andReturnSelf();
```

Comment:
Man, the guts of `\Illuminate\Database\Eloquent\Model::refresh` are *weird* but this lets us mock as if we had different data on disk than we set in memory.